src/DotVVM.Framework/Controls/HtmlWriter.cs

```diff
@@ -191,6 +202,9 @@ public void RenderSelfClosingTag(string name)
         {
             RenderBeginTagCore(name);
             writer.Write("/>");
+
+            if (this.enableWarnings && !IsSelfClosing(name))
+                Warn($"Element {name} is not self closing but is rendered as so. It may interpreted as a start tag without the end tag by the browsers.");
```

Comment:
You're missing the "be" in "be interpreted".

If start tag has indefinite article, end tag should most likely have it as well: "an end tag".

Also, the tiniest of nitpicks: "self-closing".

Reply:
Thanks, fixed that.